python/pyarrow/_csv.pyx

```diff
@@ -1355,6 +1355,10 @@ cdef class WriteOptions(_Weakrefable):
         CSV data
     delimiter : 1-character string, optional (default ",")
         The character delimiting individual cells in the CSV data.
+    eol : str, optional (default "\\n")
```

Comment:
> The tests are passing 👍
But the docstrings need an update:

Needed to escape the backslash to render in final doc.